### PR TITLE
Refactor/naming

### PR DESCRIPTION
**Pull request by arbennett**
This is mainly a cleanup on naming, but also fixes some bugs. I have removed the `apply_annual_b*` functions and simply renamed the `apply_interval_b*` functions to `apply_b*`. It rearranges some of the function parameters to be better grouped, and changes the `truth` variables to `ref` to be more in line with bias correction terminology. This also introduces the following terminology that `bmorph` should stick to:

- `raw_train_window` - the time period where the raw data will be used to calculate CDFs
- `ref_train_window` - the time period where the ref data will be used to calculate CDFs
- `apply_window` - the time period where the bias correction will actually be applied

Some of the bugs fixed are:

- Set `overlap` to days rather than years
- Use the overlap period when running `CQM`
- Marginalize only on the raw timeseries when running `CQM`

### CODE REVIEW

File: bmorph/core/workflows.py [hunk 1]
```diff
@@ -6,138 +6,36 @@
...
-        Bins for the first time series. This should only be used if `method='hist'`.
-    ybins : int, optional
-        Bins for the second time series. This should only be used if `method='hist'`.
-    rtol : float, optional
-        The desired relatie tolerance of the result for KernelDensity.
-        This should only be used if `method='kde'`.
-    atol : float, optional
-        The desired absolute tolerance of the result for KernelDensity.
-        This should only be used if `method='kde'`.
-    method : str, optional
```
Comment (steinadio) [inline bmorph/core/workflows.py:63]: it looks like method got accidentally left out of the next function and therefore cut out completely. please add it to the new `apply_bmorph` docstring

File: bmorph/core/workflows.py [hunk 2]
```diff
@@ -6,138 +6,36 @@
...
-    train_window : pandas.date_range
-        Date range to train the bias correction model.
-    bmorph_window : pandas.date_range
+    apply_window : pandas.date_range
         Date range to apply bmorph onto flow timeseries.
-    reference_window : pandas.date_range
+    raw_train_window : pandas.date_range
+        Date range to train the bias correction model.
+    ref_train_window : pandas.date_range
         Date range to smooth elements in 'raw_ts' and 'bmorph_ts'.
```
[the remark this answers is not in the document]
Comment (steinadio) [reply bmorph/core/workflows.py:32]: Should probably also put a note on what `**kwargs` are being sent where now that I think about it

File: bmorph/core/workflows.py [hunk 3]
```diff
@@ -183,260 +75,79 @@ def apply_interval_bmorph(raw_ts, train_ts, obs_ts,
...
+                                 apply_window.stop+bmorph_interval,
                                  freq=bmorph_interval)
     for i in range(0,len(bmorph_range)-1):
         bmorph_start = bmorph_range[i]
         bmorph_end = bmorph_range[i+1]
 
-        # we don't need ot adjust for overlap if it is the last entry at i+1
-        if i < len(bmorph_range)-2:
-            bmorph_end -= pd.DateOffset(days=1)
+        ## we don't need to adjust for overlap if it is the last entry at i+1
```
Comment (steinadio) [inline bmorph/core/workflows.py:103]: if this is no longer being used it should be deleted_

Comment (steinadio) [reply bmorph/core/workflows.py:103]: same for 104 and 105

File: bmorph/core/workflows.py [hunk 4]
```diff
@@ -6,138 +6,36 @@
...
-        Observed/reference flow timeseries.
-    train_window : pandas.date_range
-        Date range to train the bias correction model.
-    bmorph_window : pandas.date_range
-        Date range to apply bmorph onto flow timeseries.
-    reference_window : pandas.date_range
-        Date range to smooth elements in 'raw_ts' and 'bmorph_ts'.
     bmorph_interval : pandas.DateOffset
         Difference between bmorph application intervals.
     bmorph_overlap : int
```
Comment (steinadio) [inline bmorph/core/workflows.py:41]: to avoid the previous error, should be specified here that the overlap is in days

Comment (steinadio) [reply bmorph/core/workflows.py:41]: same for `blendmorph`

File: bmorph/core/workflows.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (steinadio) [reply bmorph/core/workflows.py:32]: some regarding `**kwargs` for `blendmorph`

**Review by steinadio — CHANGES_REQUESTED** [review]
Overall just some minor fixing docstrings. 

If you could please write a docstring for `workflows.run_parallel_scbc` that would be very helpful since it is used in the tutorial.

Regarding the tutorial, please revise the following:
- flip the axes of the SRN horizontally so that the segments better match
- make certain the font is consistent on bmorph throughout as `bmorph` or **bmorph**
- does the `-1.7e7` need to be explained when adjusting the hru's under "Loading in the streamflow data ..."
- the results plots look huge to me, should they be? (more of a comment)
- the SRN at the end appears to be squished strangely and I don't remember it before ... can look into more later

The docs need to be synced up with the new notebook code and any new naming. Please either update them or let me know if you need me to do it instead.

**Comment by arbennett** [comment]
I think I'm going to just note down the requested changes to the tutorial since that's a bit outside the scope of this PR, but will make them in a follow up PR.

File: bmorph/core/workflows.py (hunk 3 above)
Comment (arbennett) [reply bmorph/core/workflows.py]: Done

File: bmorph/core/workflows.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arbennett) [reply bmorph/core/workflows.py:37]: Done

File: bmorph/core/workflows.py (hunk 4 above)
Comment (arbennett) [reply bmorph/core/workflows.py:46]: Done

**Review by steinadio — APPROVED** [review]

**Pull request merged**